src/OmniSharp.MSBuild/ProjectSystem.cs

```diff
@@ -137,7 +137,25 @@ private IEnumerable<string> GetInitialProjectPaths()
             return _fileSystemHelper.GetFiles("**/*.csproj");
         }
 
-        private IEnumerable<string> GetProjectPathsFromSolution(string solutionFilePath)
+        // enumerating project file paths in MSBuild.SolutionSdk project.
+        private IEnumerable<string> GetProjectPathsFromSolutionBySolutionSdk(string solutionFilePath)
+        {
+            IEnumerable<string> CollectFunc(string slnproj)
+            {
+                // to avoid enumrating unsupported project.
+                foreach (var path in SolutionSdkFileUtil.GetEvaluatedProjectFilePaths(slnproj, _propertyOverrides, new string[] { ".csproj", ".fsproj", ".vbproj" }))
+                {
+                    yield return Path.IsPathRooted(path) ? path : Path.Combine(Path.GetDirectoryName(slnproj), path);
+                }
+            };
+            using (_sdksPathResolver.SetSdksPathEnvironmentVariable(solutionFilePath))
```

Comment:
please add a blank line above here to separate this code from the local function above.

Reply:
+1